hibernate-core/src/main/java/org/hibernate/loader/plan/exec/internal/AbstractLoadPlanBasedLoader.java

```diff
@@ -431,7 +431,8 @@ protected final ResultSet getResultSet(
 
 		try {
 			ResultSet rs = session.getJdbcCoordinator().getResultSetReturn().extract( st );
-			rs = wrapResultSetIfEnabled( rs , session );
+			if(session.getFactory().getSessionFactoryOptions().isWrapResultSetsEnabled())
```

Comment:
Unless this is something which can be changed at runtime, shouldn't we cache this as a final boolean field of this instance?

BTW your style is still non-hibernate; see https://github.com/hibernate/hibernate-ide-codestyles/